src/Runtime/ExecutionSession.hpp

```diff
@@ -42,5 +43,8 @@ class ExecutionSession {
 
   // Entry point function.
   entryPointFuncType _entryPointFunc = nullptr;
+
+  // Input signature function
+  inputSignatureType _inputSignatureFunc = nullptr;
```

Comment:
Since this is never used in the base class, it should not be here.

Reply:
@sstamenova I use `_inputSignatureFunc` in function `PyExecutionSession::pyInputSig`. It's just delay initializing only when it is necessary. 

```c++
py::str PyExecutionSession::pyInputSig() {
  std::string inputSignatureName = "omInputSignature";
  _inputSignatureFunc = reinterpret_cast<inputSignatureType>(
      _sharedLibraryHandle.getAddressOfSymbol(inputSignatureName.c_str()));
  assert(_inputSignatureFunc && "Input Signature func not loaded.");

  const char *outputs = _inputSignatureFunc();
  return outputs ? py::str(outputs) : "";
}
```

Since `inputSignatureType` definition is diffrent from `entryPointFuncType`, I cannot reuse `entryPointFuncType` in signature testcase.

Reply:
Your usage is not in the base class, so the declaration does not need to be in the base class. You should move the declaration to `PyExecutionSession`.